### ISSUE TITLE
Pkg up or upgrade fails at libandroid-glob_0.5_aarch64.deb (--unpack)

### ISSUE THREAD

**Issue by Fernal73** [termux/termux-packages#4150]
**Problem description**
pkg up or pkg upgrade fails

```


WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Get:1 https://its-pointless.github.io/files termux InRelease [9176 B]
Hit:2 https://termux.net stable InRelease
Ign:3 https://dl.bintray.com/grimler/game-packages-21 games InRelease
Ign:4 https://dl.bintray.com/grimler/termux-root-packages-21 root InRelease
Ign:5 https://dl.bintray.com/grimler/science-packages-21 science InRelease
Ign:6 https://dl.bintray.com/xeffyr/unstable-packages-21 unstable InRelease
Ign:7 https://dl.bintray.com/xeffyr/x11-packages-21 x11 InRelease
Get:8 https://dl.bintray.com/grimler/game-packages-21 games Release [5344 B]
Hit:8 https://dl.bintray.com/grimler/game-packages-21 games Release
Get:10 https://dl.bintray.com/grimler/termux-root-packages-21 root Release [6185 B]
Hit:10 https://dl.bintray.com/grimler/termux-root-packages-21 root Release
Get:12 https://dl.bintray.com/grimler/science-packages-21 science Release [5348 B]
Hit:12 https://dl.bintray.com/grimler/science-packages-21 science Release
Get:13 https://dl.bintray.com/xeffyr/unstable-packages-21 unstable Release [6065 B]
Hit:13 https://dl.bintray.com/xeffyr/unstable-packages-21 unstable Release
Get:16 https://dl.bintray.com/xeffyr/x11-packages-21 x11 Release [6055 B]
Hit:16 https://dl.bintray.com/xeffyr/x11-packages-21 x11 Release
Fetched 9176 B in 14s (647 B/s)
Reading package lists...
Building dependency tree...
Reading state information...
145 packages can be upgraded. Run 'apt list --upgradable' to see them.

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
The following packages will be upgraded:
  argp babl binutils binutils-gold boost c-ares clang dnsutils e2fsprogs emacs
  erlang ffmpeg fftw flex fontconfig fontconfig-utils freetype fribidi gbt gdb
  gdbm gdk-pixbuf gegl giflib giflib-utils glib glib-bin gnutls golang
  golang-doc gpgme graphviz gstreamer harfbuzz harfbuzz-icu imagemagick
  json-glib ldns ledger libandroid-glob libandroid-shmem libarchive libass
  libassuan libcaca libcroco libcrypt libdb libedit libevent libffi libflac
  libgcrypt libgd libgit2 libgmime libgnutls libgpg-error libgraphite libicu
  libid3tag libidn libidn2 libisl libjasper libjpeg-turbo libjpeg-turbo-progs
  libksba libllvm libltdl liblua liblz4 libmad libmp3lame libmsgpack libnettle
  libnpth libogg libopus libpixman libpng libprotobuf libpsl libpulseaudio
  librsvg libsndfile libsoup libsoxr libtalloc libtermkey libtiff libtool
  libunibilium libunistring libuv libvorbis libvpx libwebp libx264 libx265
  libxml2 libxml2-utils libxslt libyaml libzopfli littlecms make mariadb
  ndk-sysroot nodejs openal-soft openjpeg openjpeg-tools openssl-tool opusfile
  pcre perl php php-pgsql poppler postgresql procps protobuf python python2
  rhash ruby serf sox speexdsp subversion teckit termux-auth texlive-bin tidy
  utf8proc util-linux weechat weechat-lua-plugin weechat-perl-plugin
  weechat-python-plugin weechat-ruby-plugin xsltproc xvidcore yasm
145 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/308 MB of archives.
After this operation, 326 MB of additional disk space will be used.
Do you want to continue? [Y/n] (Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 60459 files and directories currently installed.)
Preparing to unpack .../libandroid-glob_0.5_aarch64.deb ...
Unpacking libandroid-glob (0.5) over (0.4) ...
dpkg: error processing archive /data/data/com.termux/files/usr/var/cache/apt/archives/libandroid-glob_0.5_aarch64.deb (--unpack):
 trying to overwrite '/data/data/com.termux/files/usr/include/glob.h', which is also in package libandroid-glob-dev 0.4
Errors were encountered while processing:
 /data/data/com.termux/files/usr/var/cache/apt/archives/libandroid-glob_0.5_aarch64.deb
E: Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an error code (1)
```

*Expected behavior**
UPGRADE SHOULD HAPPEN WITHOUT ANY ERROR MESSAGES. 

**Additional information**
output of command `termux-info`.

```
Updatable packages:
argp/stable 1.3-1 aarch64 [upgradable from: 1.3]
babl/stable 0.1.68-1 aarch64 [upgradable from: 0.1.68]
binutils/stable 2.32-4 aarch64 [upgradable from: 2.32-3]
binutils-gold/stable 2.32-4 aarch64 [upgradable from: 2.32-3]
boost/stable 1.70.0-5 aarch64 [upgradable from: 1.70.0-4]
c-ares/stable 1.15.0-2 aarch64 [upgradable from: 1.15.0-1]
clang/stable 8.0.0-5 aarch64 [upgradable from: 8.0.0-4]
dnsutils/stable 9.14.4-1 aarch64 [upgradable from: 9.14.4]
e2fsprogs/stable 1.45.3-1 aarch64 [upgradable from: 1.45.3]
emacs/stable 26.2-1 aarch64 [upgradable from: 26.2]
erlang/stable 21.3.8-1 aarch64 [upgradable from: 21.3.8]
ffmpeg/stable 4.1.4-2 aarch64 [upgradable from: 4.1.4-1]
fftw/stable 3.3.8-2 aarch64 [upgradable from: 3.3.8-1]
flex/stable 2.6.4-1 aarch64 [upgradable from: 2.6.4]
fontconfig/stable 2.13.1-3 aarch64 [upgradable from: 2.13.1-2]
fontconfig-utils/stable 2.13.1-3 aarch64 [upgradable from: 2.13.1-2]
freetype/stable 2.10.1-1 aarch64 [upgradable from: 2.10.1]
fribidi/stable 1.0.5-1 aarch64 [upgradable from: 1.0.5]
gbt/stable 2.0.0 aarch64 [upgradable from: 1.2.0-1]
gdb/stable 8.3-3 aarch64 [upgradable from: 8.3-2]
gdbm/stable 1.18.1-2 aarch64 [upgradable from: 1.18.1-1]
gdk-pixbuf/stable 2.38.1-2 aarch64 [upgradable from: 2.38.1-1]
gegl/stable 0.4.16-3 aarch64 [upgradable from: 0.4.16-2]
giflib/stable 5.2.1-1 aarch64 [upgradable from: 5.2.1]
giflib-utils/stable 5.2.1-1 aarch64 [upgradable from: 5.2.1]
glib/stable 2.60.6-1 aarch64 [upgradable from: 2.60.6]
glib-bin/stable 2.60.6-1 aarch64 [upgradable from: 2.60.6]
gnutls/stable 3.6.9-1 aarch64 [upgradable from: 3.6.9]
golang/stable 2:1.12.7-1 aarch64 [upgradable from: 2:1.12.7]
golang-doc/stable 2:1.12.7-1 aarch64 [upgradable from: 2:1.12.7]
gpgme/stable 1.13.1-1 aarch64 [upgradable from: 1.13.1]
graphviz/stable 2.40.1-9 aarch64 [upgradable from: 2.40.1-8]
gstreamer/stable 1.16.0-1 aarch64 [upgradable from: 1.16.0]
harfbuzz/stable 2.5.3-1 aarch64 [upgradable from: 2.5.3]
harfbuzz-icu/stable 2.5.3-1 aarch64 [upgradable from: 2.5.3]
imagemagick/stable 7.0.8.58-1 aarch64 [upgradable from: 7.0.8.58]
json-glib/stable 1.4.4-1 aarch64 [upgradable from: 1.4.4]
ldns/stable 1.7.1-1 aarch64 [upgradable from: 1.7.1]
ledger/stable 3.1.3-3 aarch64 [upgradable from: 3.1.3-2]
libandroid-glob/stable 0.5 aarch64 [upgradable from: 0.4]
libandroid-shmem/stable 0.2.1 aarch64 [upgradable from: 0.2]
libarchive/stable 3.4.0-1 aarch64 [upgradable from: 3.4.0]
libass/stable 0.14.0-1 aarch64 [upgradable from: 0.14.0]
libassuan/stable 2.5.3-1 aarch64 [upgradable from: 2.5.3]
libcaca/stable 0.99.beta19-6 aarch64 [upgradable from: 0.99.beta19-5]
libcroco/stable 0.6.13-1 aarch64 [upgradable from: 0.6.13]
libcrypt/stable 0.2-2 aarch64 [upgradable from: 0.2-1]
libdb/stable 18.1.32-2 aarch64 [upgradable from: 18.1.32-1]
libedit/stable 20190324-3.1-1 aarch64 [upgradable from: 20190324-3.1-0]
libevent/stable 2.1.11-1 aarch64 [upgradable from: 2.1.11]
libffi/stable 3.2.1-4 aarch64 [upgradable from: 3.2.1-3]
libflac/stable 1.3.2-10 aarch64 [upgradable from: 1.3.2-9]
libgcrypt/stable 1.8.4-1 aarch64 [upgradable from: 1.8.4]
libgd/stable 2.2.5-5 aarch64 [upgradable from: 2.2.5-4]
libgit2/stable 0.28.2-1 aarch64 [upgradable from: 0.28.2]
libgmime/stable 3.2.3-4 aarch64 [upgradable from: 3.2.3-3]
libgnutls/stable 3.6.9-1 aarch64 [upgradable from: 3.6.9]
libgpg-error/stable 1.36-1 aarch64 [upgradable from: 1.36]
libgraphite/stable 1.3.13-3 aarch64 [upgradable from: 1.3.13-2]
libicu/stable 64.2-2 aarch64 [upgradable from: 64.2-1]
libid3tag/stable 0.15.1b-3 aarch64 [upgradable from: 0.15.1b-2]
libidn/stable 1.35-2 aarch64 [upgradable from: 1.35-1]
libidn2/stable 2.2.0-1 aarch64 [upgradable from: 2.2.0]
libisl/stable 0.21-1 aarch64 [upgradable from: 0.21]
libjasper/stable 2.0.16-3 aarch64 [upgradable from: 2.0.16-2]
libjpeg-turbo/stable 2.0.2-3 aarch64 [upgradable from: 2.0.2-2]
libjpeg-turbo-progs/stable 2.0.2-3 aarch64 [upgradable from: 2.0.2-2]
libksba/stable 1.3.5-2 aarch64 [upgradable from: 1.3.5-1]
libllvm/stable 8.0.0-5 aarch64 [upgradable from: 8.0.0-4]
libltdl/stable 2.4.6-6 aarch64 [upgradable from: 2.4.6-5]
liblua/stable 5.3.5-3 aarch64 [upgradable from: 5.3.5-2]
liblz4/stable 1.9.1-1 aarch64 [upgradable from: 1.9.1]
libmad/stable 0.15.1b-2 aarch64 [upgradable from: 0.15.1b-1]
libmp3lame/stable 3.100-1 aarch64 [upgradable from: 3.100]
libmsgpack/stable 3.2.0-1 aarch64 [upgradable from: 3.2.0]
libnettle/stable 3.5.1-2 aarch64 [upgradable from: 3.5.1-1]
libnpth/stable 1.6-1 aarch64 [upgradable from: 1.6]
libogg/stable 1.3.3-1 aarch64 [upgradable from: 1.3.3]
libopus/stable 1.3.1-2 aarch64 [upgradable from: 1.3.1-1]
libpixman/stable 0.38.4-2 aarch64 [upgradable from: 0.38.4-1]
libpng/stable 1.6.37-2 aarch64 [upgradable from: 1.6.37-1]
libprotobuf/stable 3.9.0-1 aarch64 [upgradable from: 3.9.0]
libpsl/stable 0.21.0-1 aarch64 [upgradable from: 0.21.0]
libpulseaudio/stable 12.2-18 aarch64 [upgradable from: 12.2-17]
librsvg/stable 2.45.90-1 aarch64 [upgradable from: 2.45.90]
libsndfile/stable 1.0.29~pre1-3 aarch64 [upgradable from: 1.0.29~pre1-2]
libsoup/stable 2.66.2-1 aarch64 [upgradable from: 2.66.2]
libsoxr/stable 0.1.3-2 aarch64 [upgradable from: 0.1.3-1]
libtalloc/stable 2.1.14-2 aarch64 [upgradable from: 2.1.14-1]
libtermkey/stable 0.22-1 aarch64 [upgradable from: 0.22]
libtiff/stable 4.0.10-8 aarch64 [upgradable from: 4.0.10-7]
libtool/stable 2.4.6-6 aarch64 [upgradable from: 2.4.6-5]
libunibilium/stable 2.0.0-3 aarch64 [upgradable from: 2.0.0-2]
libunistring/stable 0.9.10-3 aarch64 [upgradable from: 0.9.10-2]
libuv/stable 1.30.1-1 aarch64 [upgradable from: 1.30.1]
libvorbis/stable 1.3.6-2 aarch64 [upgradable from: 1.3.6-1]
libvpx/stable 1.8.1-1 aarch64 [upgradable from: 1.8.1]
libwebp/stable 1.0.3-1 aarch64 [upgradable from: 1.0.3]
libx264/stable 20190215-1 aarch64 [upgradable from: 20190215]
libx265/stable 3.1.1-1 aarch64 [upgradable from: 3.1.1]
libxml2/stable 2.9.9-4 aarch64 [upgradable from: 2.9.9-3]
libxml2-utils/stable 2.9.9-4 aarch64 [upgradable from: 2.9.9-3]
libxslt/stable 1.1.33-2 aarch64 [upgradable from: 1.1.33-1]
libyaml/stable 0.2.2-1 aarch64 [upgradable from: 0.2.2]
libzopfli/stable 1.0.2-4 aarch64 [upgradable from: 1.0.2-3]
littlecms/stable 2.9-1 aarch64 [upgradable from: 2.9]
make/stable 4.2.1-3 aarch64 [upgradable from: 4.2.1-2]
mariadb/stable 10.4.6-2 aarch64 [upgradable from: 10.4.6]
ndk-sysroot/stable 20-1 aarch64 [upgradable from: 20]
nodejs/stable 12.4.0-3 aarch64 [upgradable from: 12.4.0-2]
openal-soft/stable 1.19.1-1 aarch64 [upgradable from: 1.19.1]
openjpeg/stable 2.3.1-1 aarch64 [upgradable from: 2.3.1]
openjpeg-tools/stable 2.3.1-1 aarch64 [upgradable from: 2.3.1]
openssl-tool/stable 1.1.1c-2 aarch64 [upgradable from: 1.1.1c-1]
opusfile/stable 0.11-2 aarch64 [upgradable from: 0.11-1]
pcre/stable 8.43-3 aarch64 [upgradable from: 8.43-2]
perl/stable 5.30.0-1 aarch64 [upgradable from: 5.30.0]
php/stable 7.3.7-1 aarch64 [upgradable from: 7.3.7]
php-pgsql/stable 7.3.7-1 aarch64 [upgradable from: 7.3.7]
poppler/stable 0.79.0-1 aarch64 [upgradable from: 0.79.0]
postgresql/stable 11.4-1 aarch64 [upgradable from: 11.4]
procps/stable 3.3.15-3 aarch64 [upgradable from: 3.3.15-2]
protobuf/stable 3.9.0-1 aarch64 [upgradable from: 3.9.0]
python/stable 3.7.4-1 aarch64 [upgradable from: 3.7.4]
python2/stable 2.7.16-3 aarch64 [upgradable from: 2.7.16-2]
rhash/stable 1.3.8-2 aarch64 [upgradable from: 1.3.8-1]
ruby/stable 2.6.3-5 aarch64 [upgradable from: 2.6.3-4]
serf/stable 1.3.9-4 aarch64 [upgradable from: 1.3.9-3]
sox/stable 14.4.2-13 aarch64 [upgradable from: 14.4.2-12]
speexdsp/stable 1.2.0-1 aarch64 [upgradable from: 1.2.0]
subversion/stable 1.12.2-1 aarch64 [upgradable from: 1.12.2]
teckit/stable 2.5.9-3 aarch64 [upgradable from: 2.5.9-2]
termux-auth/stable 1.1-1 aarch64 [upgradable from: 1.1]
texlive-bin/stable 20190410-5 aarch64 [upgradable from: 20190410-4]
tidy/stable 5.7.28-1 aarch64 [upgradable from: 5.7.28]
utf8proc/stable 2.4.0-1 aarch64 [upgradable from: 2.4.0]
util-linux/stable 2.34-2 aarch64 [upgradable from: 2.34-1]
weechat/stable 2.5-1 aarch64 [upgradable from: 2.5]
weechat-lua-plugin/stable 2.5-1 aarch64 [upgradable from: 2.5]
weechat-perl-plugin/stable 2.5-1 aarch64 [upgradable from: 2.5]
weechat-python-plugin/stable 2.5-1 aarch64 [upgradable from: 2.5]
weechat-ruby-plugin/stable 2.5-1 aarch64 [upgradable from: 2.5]
xsltproc/stable 1.1.33-2 aarch64 [upgradable from: 1.1.33-1]
xvidcore/stable 1.3.5-1 aarch64 [upgradable from: 1.3.5]
yasm/stable 1.3.0-2 aarch64 [upgradable from: 1.3.0-1]
Subscribed repositories:
https://dl.bintray.com/xeffyr/x11-packages-21 x11/main
https://dl.bintray.com/xeffyr/unstable-packages-21 unstable/main
https://dl.bintray.com/grimler/science-packages-21 science/stable
https://dl.bintray.com/grimler/termux-root-packages-21 root/stable
https://its-pointless.github.io/files termux/extras
https://its-pointless.github.io/files termux/extras
https://dl.bintray.com/grimler/game-packages-21 games/stable
https://termux.net stable/main
https://termux.net stable/main
System information:
Linux localhost 3.18.31-perf-ge8c6508 #1 SMP PREEMPT Fri Jan 18 23:42:34 WIB 2019 aarch64 Android
Termux-packages arch:
aarch64
Android version:
7.1.2
```

**Comment by Grimler91** [comment]
Workaround before it is fixed: `apt remove libandroid-glob-dev; pkg upgrade`

**Comment by xeffyr** [comment]
Error happens because some updates are not available on **https://termux.net** repository. Wait until @fornwall will rebuild packages.

And if you are on "Android version: 7.1.2", I really suggest to move on Android 7+ repository - updates/fixes here delivered much faster.

**Comment by Fernal73** [comment]
Thanks. Got it.
How do I move to Android 7+?

**Comment by xeffyr** [comment]
Yes, but you can use script `termux-upgrade-repo` and follow its instructions.
In both cases if you have custom configs, databases, other **custom** things inside `$PREFIX` - backup them.

**Comment by Fernal73** [comment]
I have Arch Linux on Termux installed. That's in my home directory. Any specific instructions for that?

**Comment by xeffyr** [comment]
> Any other advantages besides faster access to updates?

Everything is compiled against newer libc (has less restrictions on available functions) and uses features available in newer Android linker such as `DT_RUNPATH` and versioning. Fixes some issues when using libraries from both Termux and /system/lib and allows to drop LD_LIBRARY_PATH.

> That's in my home directory

Home directory is untouched, only $PREFIX will be wiped.
There also reinstallation script will be generated and placed to $HOME.

**Comment by Fernal73** [comment]
Thanks a lot. Much appreciated.

**Comment by xeffyr** [comment]
Should be fixed with libandroid-glob v0.6 - should now replace devpackage without errors.
